agenta-cli/agenta/sdk/decorators/llm_entrypoint.py

```diff
@@ -136,36 +182,44 @@ def __init__(self, func: Callable[..., Any], route_path=""):
         async def wrapper(*args, **kwargs) -> Any:
             func_params, api_config_params = self.split_kwargs(kwargs, config_params)
             self.ingest_files(func_params, ingestible_files)
-            ag.config.set(**api_config_params)
+            if not config_schema:
+                ag.config.set(**api_config_params)
 
             # Set the configuration and environment of the LLM app parent span at run-time
             ag.tracing.update_baggage(
                 {"config": config_params, "environment": "playground"}
             )
-
-            entrypoint_result = await self.execute_function(
-                func,
-                True,  # inline trace: True
-                *args,
-                params=func_params,
-                config_params=config_params,
-            )
+            with route_context_manager(config=api_config_params):
+                entrypoint_result = await self.execute_function(
+                    func,
+                    True,  # inline trace: True
+                    *args,
+                    params=func_params,
+                    config_params=config_params,
+                )
 
             return entrypoint_result
 
         self.update_function_signature(
-            wrapper, func_signature, config_params, ingestible_files
+            wrapper=wrapper,
+            func_signature=func_signature,
+            config_class=config,
+            config_dict=config_params,
+            ingestible_files=ingestible_files,
         )
 
         #
-        if route_path == "":
+        if route_path == "" or route_path == "/":
```

Comment:
This is only needed because of the change in `class route(...)` which doesn't add anything, does it ?